crates/std_detect/src/detect/os/linux/s390x.rs

```diff
@@ -76,57 +68,77 @@ impl From<auxvec::AuxVec> for AtHwcap {
+                "stfle 0({})",
+                in(reg_addr) result.as_mut_ptr() ,
+                inout("r0") result.len() as u64 - 1 => _,
+                options(nostack)
+            );
+        }
+        Self(result)
+    }
 
-            // others
+    const fn get_bit(&self, n: usize) -> bool {
+        // NOTE: bits are numbered from the left.
+        self.0[n / 64] & (1 << (63 - (n % 64))) != 0
+    }
+}
 
-            // bit 45 of the extended facility list
-            enable_feature(Feature::high_word, self.high_gprs);
+/// Initializes the cache from the feature bits.
+///
+/// These values are part of the platform-specific [asm/elf.h][kernel], and are a selection of the
+/// fields found in the [Facility Indications].
+///
+/// [Facility Indications]: https://www.ibm.com/support/pages/sites/default/files/2021-05/SA22-7871-10.pdf#page=63
+/// [kernel]: https://github.com/torvalds/linux/blob/b62cef9a5c673f1b8083159f5dc03c1c5daced2f/arch/s390/include/asm/elf.h#L129
+fn cache(hwcap: Option<AtHwcap>, facilities: ExtendedFacilityList) -> cache::Initializer {
+    let mut value = cache::Initializer::default();
+
+    {
+        let mut enable_if_set = |bit_index, f| {
+            if facilities.get_bit(bit_index) {
+                value.set(f as u32);
+            }
+        };
+
+        // We use HWCAP for `vector` because it requires both hardware and kernel support.
+        if let Some(AtHwcap { vxrs: true, .. }) = hwcap {
+            // vector and related
 
-            // bit 73 of the extended facility list
-            enable_feature(Feature::transactional_execution, self.te);
+            enable_if_set(129, Feature::vector);
 
-            // bit 133 of the extended facility list
-            enable_feature(Feature::guarded_storage, self.gs);
+            enable_if_set(135, Feature::vector_enhancements_1);
+            enable_if_set(148, Feature::vector_enhancements_2);
+            enable_if_set(198, Feature::vector_enhancements_3);
 
-            // bit 150 of the extended facility list
-            enable_feature(Feature::enhanced_sort, self.sort);
+            enable_if_set(134, Feature::vector_packed_decimal);
+            enable_if_set(152, Feature::vector_packed_decimal_enhancement);
+            enable_if_set(192, Feature::vector_packed_decimal_enhancement_2);
+            enable_if_set(199, Feature::vector_packed_decimal_enhancement_3);
 
-            // bit 151 of the extended facility list
-            enable_feature(Feature::deflate_conversion, self.dflt);
+            enable_if_set(165, Feature::nnp_assist);
         }
-        value
+
+        // others
+
+        enable_if_set(45, Feature::high_word);
+        enable_if_set(73, Feature::transactional_execution);
+        enable_if_set(133, Feature::guarded_storage);
+        enable_if_set(150, Feature::enhanced_sort);
+        enable_if_set(151, Feature::deflate_conversion);
+        // added in z17
```

Comment:
Do we need this comment?  If so, should we add this information also for the vector-related facilities, and also specific machine information for pre-z17 machines?

Reply:
nah, this list doesn't really have structure to it, but it should be OK given the official docs.